### PR TITLE
Fix configuration

### PR DESCRIPTION
**Pull request by oak11**
@almet @Natim  @leplatrem

### CODE REVIEW

File: kinto/__main__.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 from kinto.config import init
 
-CONFIG_FILE = 'config/kinto.ini'
+CONFIG_FILE = 'kinto/config/kinto.ini'
```
Comment (almet) [inline kinto/__main__.py]: Why do you need to specify the kinto prefix here?

Comment (oak11) [inline kinto/__main__.py]: so that the kinto.ini file is created within the kinto/config folder( where kinto.tpl is present) . If only config/kinto.ini is written, it creates a new config folder in the root directory and kinto.ini is created inside that.

Comment (almet) [inline kinto/__main__.py]: I believe that's actually the behaviour we want: the "kinto" folder is the one where all the kinto files are contained, and you shouldn't temper it. Rather, creating a "config" folder where you run the command seems to make more sense.

Comment (oak11) [inline kinto/__main__.py]: oh, okay I'll change it back to config/kinto.ini

File: kinto/__main__.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def main(args=None):
         if os.path.exists(config_file):
             print("%s already exist." % config_file, file=sys.stderr)
             sys.exit(1)
-
+
```
Comment (almet) [inline kinto/__main__.py]: Please remove this empty line :)

Comment (oak11) [inline kinto/__main__.py]: done

**Comment by almet** [comment]
Okay, great. Now we have to write the tests!

File: kinto/config/__init__.py [hunk 3]
```diff
@@ -1,46 +1,46 @@
...
 HERE = os.path.abspath(os.path.dirname(__file__))
 
+def render_template(template, destination, values):
+   
+   here = os.path.abspath(os.path.dirname(__file__))
+   template = os.path.join(here, 'kinto.tpl') 
+   destination = os.path.join(here, 'kinto.ini')
 
-def render_template(template, destination, **kwargs):
-    template = os.path.join(HERE, template)
```
Comment (Natim) [inline kinto/config/__init__.py]: Please keep this version with the makedirs and using the HERE variable.

File: kinto/config/__init__.py [hunk 4]
```diff
@@ -1,46 +1,46 @@
...
 
     else:
-        values['storage_url'] = ''
-        values['cache_url'] = ''
-        values['permission_url'] = ''
+        values.update({'storage_url' : '',
+                     'cache_url' : '',
+                     'permission_url' : ''})
 
-    render_template("kinto.tpl", config_file, **values)
```
Comment (Natim) [inline kinto/config/__init__.py]: Why is this change necessary?

**Comment by oak11** [comment]
Cool, tell me the locations to put tests for kinto/config/__init_.py  and  kinto/__main__.py  I'll write the tests and commit them.

File: kinto/config/__init__.py (hunk 4 above)
Comment (almet) [inline kinto/config/__init__.py]: We thought it was actually better to pass the variables as a dict rather than as parameters.

**Comment by almet** [comment]
Tests for config should go in `kinto/tests/test_config.py` and don't bother about having tests for `kinto.__main__` for now.

File: kinto/config/__init__.py (hunk 4 above)
Comment (Natim) [inline kinto/config/__init__.py]: I don't agree, why would I write:

    render_template("kinto.tpl", config_file, {"secret": "abc", "storage_backend": "bla", ...})

Isn't it better to write:

    render_template("kinto.tpl", config_file, secret="abc", storage_backend="bla", ...)

File: kinto/config/__init__.py (hunk 3 above)
Comment (almet) [inline kinto/config/__init__.py]: yes, the only think you might want to change here is the mode which should be "w" instead of "w+".

File: kinto/config/__init__.py (hunk 4 above)
Comment (almet) [inline kinto/config/__init__.py]: Both solutions work for me, and I don't consider this really important. I've see with Shweta that the version without the kwargs would make more sense (be more explicit), but we can reverse it to the previous behaviour if you want.

File: kinto/tests/test_config.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+from ..config import render_template as origin
```
Comment (almet) [inline kinto/tests/test_config.py]: You need to import with absolute locations. `from kinto.config import render_template`. You don't need to import it as origin.

File: kinto/tests/test_config.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+from .support import unittest
+
+class ConfigTest(unittest.TestCase):
+
+#to check if values dict is not empty
```
Comment (almet) [inline kinto/tests/test_config.py]: Please define the comments from inside the method, and not on top of it. eg.

```python

def test_values_is_not_empty(self):
    # Your comment here.
```

That being said, your comment here just repeats the name of the method, so your don't need to add one.

File: kinto/tests/test_config.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    def test_values_is_not_empty(self):
+        self.assertNotEqual(origin.values, {})
```
Comment (almet) [inline kinto/tests/test_config.py]: It's not clear to me what the intent is here. What are you trying to test?

Comment (oak11) [inline kinto/tests/test_config.py]: I am trying to see if values has elements. If it has elements (is not empty) then it is correct- as it will have elements only if it has passed through the if statements and init function. 
If it is empty (has no elements), there was a failure somewhere.

**Comment by Natim** [comment]
Superseeded by #349
